packages/web3torrent/src/library/web3torrent-lib.ts

```diff
@@ -519,18 +523,18 @@ export default class WebTorrentPaidStreamingClient extends WebTorrent {
         const {seedingChannelId, leechingChannelId} = wire.paidStreamingExtension;
         if (leechingChannelId) {
           log.info(`About to close payment channel ${leechingChannelId}`);
-          await this.paymentChannelClient.getLatestPaymentReceipt(leechingChannelId);
-          await this.paymentChannelClient.closeChannel(leechingChannelId);
-          wire.paidStreamingExtension.leechingChannelId = null;
-          log.info(`Channel ${leechingChannelId} closed`);
+          return this.paymentChannelClient.closeChannel(leechingChannelId).then(() => {
+            wire.paidStreamingExtension.leechingChannelId = null;
+            log.info(`Channel closed: ${leechingChannelId}`);
+          });
```

Comment:
Why the switch to `.then`?

Reply:
I wanted to chain promises, but I'll make an extra function and use that instead, so that I maintain consistency.

Reply:
I'm curious why you wanted to chain promises?

Reply:
https://github.com/statechannels/monorepo/pull/1969/commits/034e40f790457b167989f3792ebeda37eaecdbb4

Reply:
> I'm curious why you wanted to chain promises?

Because the seeding or leeching `channelId` should only be closed AFTER the channel has been closed (because it's used by the communication flow). By chaining promises, we can launch `channelClose` for all wires at the same time (with an array and using the Promise.all), which results in a faster torrent cancelling.

Reply:
Hmm, I'm not sure I follow. Two awaits in a row is semantically equivalent to one promise plus one `then`, isn't it?

Reply:
IIRC no:
```
await Promise()
await Promise()
```
is not equivalent to
```
await Promise().then(()=>Promise())
```
but to
```
async ()=>{
  await Promise()
  await Promise()
}
```

And for a `Promise.all` I understand that you need an array of single promises...
..
...
now that I think about it, in this case, you might be right... well, it's been changed already.